src/parsers/manifest/dash/common/indexes/timeline/parse_s_element.ts

```diff
@@ -32,36 +34,38 @@ export interface IParsedS {
 
 /**
  * Parse a given <S> element in the MPD into a JS Object.
- * @param {Element} root
+ * @param {Object} root
  * @returns {Object}
  */
-export default function parseSElement(root: Element): IParsedS {
+export default function parseSElement(root: ITNode): IParsedS {
   const parsedS: IParsedS = {};
 
-  for (let j = 0; j < root.attributes.length; j++) {
-    const attribute = root.attributes[j];
-
-    switch (attribute.name) {
+  for (const attributeName of Object.keys(root.attributes)) {
+    const attributeVal = root.attributes[attributeName];
```

Comment:
This can also be done like that `for (const [key, value] of Object.entries(object1)) {`

Reply:
Keep it like that if you think it's more consistent with what is done somewhere else